src/View/IconView.vala

```diff
@@ -34,69 +34,53 @@ namespace Files {
+            tree.selection_changed.connect (on_view_selection_changed);
+        }
 
-            name_renderer = new Files.TextRenderer (ViewMode.ICON);
-            icon_renderer = new Files.IconRenderer (ViewMode.ICON);
+        protected override void disconnect_tree_signals () {
+            tree.selection_changed.disconnect (on_view_selection_changed);
+        }
 
-            set_up_name_renderer ();
+        protected override Gtk.Widget? create_view () {
+            tree = new Gtk.IconView.with_model (model) {
+                selection_mode = MULTIPLE,
+                columns = -1,
+            };
 
+            icon_renderer = new Files.IconRenderer (ViewMode.ICON);
             tree.pack_start (icon_renderer, false);
-            tree.pack_end (name_renderer, false);
+            tree.add_attribute (icon_renderer, "file", ListModel.ColumnID.FILE_COLUMN);
 
+            name_renderer = new Files.TextRenderer (ViewMode.ICON) {
+                wrap_mode = WORD_CHAR,
+                xalign = 0.5f,
+                yalign = 0.0f
+            };
+            set_up_name_renderer ();
+            tree.pack_end (name_renderer, false);
             tree.add_attribute (name_renderer, "text", ListModel.ColumnID.FILENAME);
             tree.add_attribute (name_renderer, "file", ListModel.ColumnID.FILE_COLUMN);
             tree.add_attribute (name_renderer, "background", ListModel.ColumnID.COLOR);
-            tree.add_attribute (icon_renderer, "file", ListModel.ColumnID.FILE_COLUMN);
 
             connect_tree_signals ();
             tree.realize.connect ((w) => {
                 tree.grab_focus ();
             });
-        }
-
-        protected override void set_up_name_renderer () {
-            base.set_up_name_renderer ();
-            name_renderer.wrap_mode = Pango.WrapMode.WORD_CHAR;
-            name_renderer.xalign = 0.5f;
-            name_renderer.yalign = 0.0f;
-        }
-
-        protected override void connect_tree_signals () {
-            tree.selection_changed.connect (on_view_selection_changed);
-        }
-
-        protected override void disconnect_tree_signals () {
-            tree.selection_changed.disconnect (on_view_selection_changed);
-        }
-
-        protected override Gtk.Widget? create_view () {
-            tree = new Gtk.IconView ();
-            set_up_view ();
 
             return tree as Gtk.Widget;
         }
 
         protected override void set_up_zoom_level () {
+            minimum_zoom = (ZoomLevel)Files.icon_view_settings.get_enum ("minimum-zoom-level");
+            maximum_zoom = (ZoomLevel)Files.icon_view_settings.get_enum ("maximum-zoom-level");
+            zoom_level = (ZoomLevel)Files.icon_view_settings.get_enum ("zoom-level");
+
             Files.icon_view_settings.bind (
                 "zoom-level",
                 this, "zoom-level",
-                GLib.SettingsBindFlags.DEFAULT
+                GLib.SettingsBindFlags.SET
```

Comment:
Should this be `SET | SYNC_CREATE`?

Reply:
That is for property binding  SettingsBindFlags.SYNC_CREATE does not exist.  The previous line initialises `zoom-level` the setting.

Reply:
Oh sorry I misread my bad!